interface/drugs/add_edit_drug.php

```diff
@@ -176,55 +182,50 @@ function sel_related() {
+            "allow_combining" => (empty($_POST['form_allow_combining']) ? 0 : 1),
+            "active" => (empty($_POST['form_active']) ? 0 : 1)      
+           );
+           $updateDrug_id = $drugService->updateDrug($updateDrug, $drug_id);
+
         } else { // deleting
             if (acl_check('admin', 'super')) {
-                sqlStatement("DELETE FROM drug_inventory WHERE drug_id = ?", array($drug_id));
-                sqlStatement("DELETE FROM drug_templates WHERE drug_id = ?", array($drug_id));
-                sqlStatement("DELETE FROM drugs WHERE drug_id = ?", array($drug_id));
-                sqlStatement("DELETE FROM prices WHERE pr_id = ? AND pr_selector != ''", array($drug_id));
+               $delete_drug_id = $drugService->deleteDrug($drug_id);
             }
         }
     } else if ($_POST['form_save']) { // saving a new drug
         $new_drug = true;
-        $drug_id = sqlInsert("INSERT INTO drugs ( " .
-        "name, ndc_number, drug_code, on_order, reorder_point, max_level, form, " .
-        "size, unit, route, cyp_factor, related_code, " .
-        "allow_multiple, allow_combining, active " .
-        ") VALUES ( " .
-        "'" . escapedff('form_name')          . "', " .
-        "'" . escapedff('form_ndc_number')    . "', " .
-        "'" . escapedff('form_drug_code')    . "', " .
-        "'" . escapedff('form_on_order')      . "', " .
-        "'" . escapedff('form_reorder_point') . "', " .
-        "'" . escapedff('form_max_level')     . "', " .
-        "'" . escapedff('form_form')          . "', " .
-        "'" . escapedff('form_size')          . "', " .
-        "'" . escapedff('form_unit')          . "', " .
-        "'" . escapedff('form_route')         . "', " .
-        "'" . numericff('form_cyp_factor')    . "', " .
-        "'" . escapedff('form_related_code')  . "', " .
-        (empty($_POST['form_allow_multiple' ]) ? 0 : 1) . ", " .
-        (empty($_POST['form_allow_combining']) ? 0 : 1) . ", " .
-        (empty($_POST['form_active']) ? 0 : 1)        .
-        ")");
+           $addDrug = array(
+            "name" => escapedff('form_name'),
+            "ndc_number" => escapedff('form_ndc_number'),
+            "drug_code" => escapedff('form_drug_code'),
+            "on_order" => escapedff('form_on_order'),
+            "reorder_point" => escapedff('form_reorder_point'),
+            "max_level" => escapedff('form_max_level'),
+            "form" => escapedff('form_form'),
+            "size" => escapedff('form_sixe'),
+            "unit" => escapedff('form_unit'),
+            "route" => escapedff('form_route'),
+            "cyp_factor" => numericff('form_cyp_factor'),
+            "related_code" => escapedff('form_related_code'),
+            "allow_multiple" => (empty($_POST['form_allow_multiple' ]) ? 0 : 1),
+            "allow_combining" => (empty($_POST['form_allow_combining']) ? 0 : 1),
+            "active" => (empty($_POST['form_active']) ? 0 : 1)      
```

Comment:
remove all the escapedff (and change the numericff to a trim()) and instead use sql binding in the function in the service class.